compiler/src/main/java/com/squareup/anvil/compiler/ModuleMerger.kt

```diff
@@ -247,21 +232,21 @@ internal class ModuleMerger(
     if (predefinedModules != null) {
       val intersect = predefinedModules.intersect(excludedModules.toSet())
       if (intersect.isNotEmpty()) {
-        throw AnvilCompilationException(
-          thisDescriptor,
-          "${thisDescriptor.name} includes and excludes modules " +
-            "at the same time: ${intersect.joinToString { it.className }}"
+        throw AnvilCompilationExceptionClassReference(
+          clazz,
+          "${clazz.clazz.name} includes and excludes modules " +
+            "at the same time: ${intersect.joinToString { it.className.toString() }}"
         )
       }
     }
 
     val contributedSubcomponentModules =
-      findContributedSubcomponentModules(thisDescriptor, scopeFqName, module)
+      findContributedSubcomponentModules(clazz, scopeFqName, module)
         .map { codegen.typeMapper.mapType(it) }
 
     val contributedModules = modules
       .asSequence()
-      .map { codegen.typeMapper.mapType(it.first) }
+      .map { codegen.typeMapper.mapType(it.first.clazz) }
       .minus(replacedModules.toSet())
       .minus(replacedModulesByContributedBindings.toSet())
       .minus(replacedModulesByContributedMultibindings.toSet())
```

Comment:
I think I can answer my own question. Here we need this other `Type`. But I think until this point we could fully work with `ClassReference` and then below convert them to this asm type.

Reply:
Hmm I had originally converted everything to fully use ClassReference and went back because it looked like the conversion couldn't be done (and I had been missing this piece until the very end since it got deleted early). I think you're right that it would work and I could just use this method for each set here at the end. I'll give that a shot.